### PR TITLE
Create bug-bounty-preview-scopes.md

### PR DESCRIPTION
**Pull request by blowdart**
Create initial list of features which, whilst preview, are still in scope for the bug bounty.

This file will be linked from the main bounty page via an aka.ms linke.

### CODE REVIEW

File: bug-bounty-preview-scopes.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
+# .NET Bounty Scope
+
+## Introduction
+
+The [.NET bug bounty](https://www.microsoft.com/en-us/msrc/bounty-dot-net-core) covers current released versions of .NET Core, ASP.NET Core and .NET, as well as the latest beta or release candidate of the upcoming version.
+
+.NET has the concept of preview features which ship in the current version, or in nightly builds of the upcoming version. Preview features are not enabled by default, but their goal is to be enabled in the next major release. Preview features are now considered in scope for the bug bounty if they are listed in the table below.
```
Comment (terrajobst) [inline bug-bounty-preview-scopes.md:7]:
```suggestion
.NET has the concept of [preview features](https://github.com/dotnet/designs/blob/main/accepted/2021/preview-features/preview-features.md) which ship in the current version, or in nightly builds of the upcoming version. Preview features are not enabled by default, but their goal is to be enabled in the next major release. Preview features are now considered in scope for the bug bounty if they are listed in the table below.
```

File: bug-bounty-preview-scopes.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+Bugs against preview features in the current release must be demonstrated against the latest current release.
+
+Bugs against preview features in nightly builds of the next version must be demonstrated against the build number listed in the table below, or subsequent builds. Preview features in nightly builds may fall out of scope while bugs are addressed or the removal of the feature from the upcoming version.
```
Comment (terrajobst) [inline bug-bounty-preview-scopes.md:11]: > Bugs against preview features in nightly builds of the next version must be demonstrated against the build number listed in the table below, or subsequent builds.

How often do you plan to update this table? Also, maybe I'm missing something, but why are the rules for nightlies different from current releases (latest only)?

File: bug-bounty-preview-scopes.md [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+Any inclusion of a feature in the table below should not be taken as an indication that the feature will ship in any upcoming version.
+
+Documenation security bugs for preview features are only in scope for the current released version of .NET
```
Comment (terrajobst) [inline bug-bounty-preview-scopes.md:15]:
```suggestion
Documenation of security bugs for preview features are only in scope for the current released version of .NET
```

File: bug-bounty-preview-scopes.md [hunk 4]
```diff
@@ -0,0 +1,33 @@
+# .NET Bounty Scope
```
Comment (terrajobst) [inline bug-bounty-preview-scopes.md:1]: The bug bounty seems to be scoped to security features. Should the title reflect that?

File: bug-bounty-preview-scopes.md (hunk 2 above)
Comment (blowdart) [reply bug-bounty-preview-scopes.md:11]: Because nightlies have been excluded because they won't have undergone security reviews and contain half-baked stuff there for testing. This now includes features which we consider complete and ready for beating up, even if they're in nightlies, but limits the scope to those nightlies, keeping legal happy :)

File: bug-bounty-preview-scopes.md (hunk 4 above)
Comment (blowdart) [reply bug-bounty-preview-scopes.md:1]: Legal :) But yea, the bounty only covers things that would create an MSRC, that's just the rules.

**Review by terrajobst — APPROVED** [review]

File: bug-bounty-preview-scopes.md [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+
+.NET has the concept of [preview features](https://github.com/dotnet/designs/blob/main/accepted/2021/preview-features/preview-features.md) which ship in the current version, or in nightly builds of the upcoming version. Preview features are not enabled by default, but their goal is to be enabled in the next major release. Preview features are now considered in scope for the bug bounty if they are listed in the table below.
+
+Bugs against preview features in the current release must be demonstrated against the latest current release.
+
+Bugs against preview features in nightly builds of the next version must be demonstrated against the build number listed in the table below, or subsequent builds. Preview features in nightly builds may fall out of scope while bugs are addressed or the removal of the feature from the upcoming version.
+
+Any inclusion of a feature in the table below should not be taken as an indication that the feature will ship in any upcoming version.
+
+Documenation of security bugs for preview features are only in scope for the current released version of .NET
```
Comment (mairaw) [inline bug-bounty-preview-scopes.md:15]:
```suggestion
Documentation of security bugs for preview features are only in scope for the current released version of .NET
```

File: bug-bounty-preview-scopes.md [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+
+Please read the .NET bug bounty [terms and conditions](https://www.microsoft.com/en-us/msrc/bounty-dot-net-core) before submitting your bug report. To be eligible for a bounty you must submit your bug report through the Microsoft Security Response Center (MSRC).
```
Comment (mairaw) [inline bug-bounty-preview-scopes.md:17]: Cam we be more direct here?

```suggestion
Read the .NET bug bounty [terms and conditions](https://www.microsoft.com/en-us/msrc/bounty-dot-net-core) before submitting your bug report. To be eligible for a bounty, you must submit your bug report through the Microsoft Security Response Center (MSRC).
```

File: bug-bounty-preview-scopes.md [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+
+## In scope preview features in the current release (.NET 6)
+
+| Feature | Description | Documentation |
+|---------|-------------|---------------|
+| HTTP/3 | HTTP/3 support in the Kestrel Web Server | [Enabling HTTP/3](https://docs.microsoft.com/en-us/aspnet/core/fundamentals/servers/kestrel/http3?view=aspnetcore-6.0) |
```
Comment (mairaw) [inline bug-bounty-preview-scopes.md:23]:
```suggestion
| HTTP/3 | HTTP/3 support in the Kestrel Web Server | [Enabling HTTP/3](https://docs.microsoft.com/aspnet/core/fundamentals/servers/kestrel/http3?view=aspnetcore-6.0) |
```

File: bug-bounty-preview-scopes.md [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+
+Bugs against preview features in nightly builds of the next version must be demonstrated against the build number listed in the table below, or subsequent builds. Preview features in nightly builds may fall out of scope while bugs are addressed or the removal of the feature from the upcoming version.
+
+Any inclusion of a feature in the table below should not be taken as an indication that the feature will ship in any upcoming version.
+
+Documenation of security bugs for preview features are only in scope for the current released version of .NET
+
+Please read the .NET bug bounty [terms and conditions](https://www.microsoft.com/en-us/msrc/bounty-dot-net-core) before submitting your bug report. To be eligible for a bounty you must submit your bug report through the Microsoft Security Response Center (MSRC).
+
+## In scope preview features in the current release (.NET 6)
```
Comment (mairaw) [inline bug-bounty-preview-scopes.md:19]:
```suggestion
## In-scope preview features in the current release (.NET 6)
```

File: bug-bounty-preview-scopes.md [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+
+| Feature | Description | Documentation |
+|---------|-------------|---------------|
+| HTTP/3 | HTTP/3 support in the Kestrel Web Server | [Enabling HTTP/3](https://docs.microsoft.com/en-us/aspnet/core/fundamentals/servers/kestrel/http3?view=aspnetcore-6.0) |
+
+## In scope preview features in nightly builds of the upcoming version (.NET 7)
```
Comment (mairaw) [inline bug-bounty-preview-scopes.md:25]:
```suggestion
## In-scope preview features in nightly builds of the upcoming version (.NET 7)
```

File: bug-bounty-preview-scopes.md [hunk 10]
```diff
@@ -0,0 +1,33 @@
+# .NET Bounty Scope
+
+## Introduction
+
+The [.NET bug bounty](https://www.microsoft.com/en-us/msrc/bounty-dot-net-core) covers current released versions of .NET Core, ASP.NET Core and .NET, as well as the latest beta or release candidate of the upcoming version.
+
+.NET has the concept of [preview features](https://github.com/dotnet/designs/blob/main/accepted/2021/preview-features/preview-features.md) which ship in the current version, or in nightly builds of the upcoming version. Preview features are not enabled by default, but their goal is to be enabled in the next major release. Preview features are now considered in scope for the bug bounty if they are listed in the table below.
+
```
Comment (GrabYourPitchforks) [inline bug-bounty-preview-scopes.md:8]: Let's assume for the sake of argument that HTTP/3 support ships proper in .NET 7. How would the "current" release play out in the table below? Do we say that as soon as the feature goes stable and vNext ships, it's now unsupported (thus ineligible for bounty) in .NET 6?

Comment (terrajobst) [reply bug-bounty-preview-scopes.md:8]: Nit: Technically, HTTP/3 is always unsupported in .NET 6, even before .NET 7 ships (it's marked as preview).

I think what you mean to ask is: do preview features in the bug bounty program have the same servicing lifespan as the rest of release? I'd say yes, otherwise it would be hard to follow.

Comment (blowdart) [reply bug-bounty-preview-scopes.md:8]: That's an interesting question, I don't know. It would depend on what we discovered, but I'd presume that we backport fixes to any preview in an LTS release. This is more of an internal process question though.

File: bug-bounty-preview-scopes.md (hunk 6 above)
Comment (blowdart) [reply bug-bounty-preview-scopes.md:17]: Goodness, not often I'm called out for being polite :p

Comment (mairaw) [reply bug-bounty-preview-scopes.md]: LOL - it's Microsoft style guide unless you think this is an inconvenient ask 😄 
https://docs.microsoft.com/en-us/style-guide/a-z-word-list-term-collections/p/please

**Review by GrabYourPitchforks — APPROVED** [review]

File: bug-bounty-preview-scopes.md (hunk 10 above)
Comment (terrajobst) [reply bug-bounty-preview-scopes.md:8]: The challenge is that some of the preview support might require tooling changes (e.g. in the SDK or compiler). For .NET 7 we might have reacted to customer feedback from .NET 6 that requires making changes in the tooling. Since we don't want to support both, the old way and the new way in the .NET 7 SDK, we might not be able to enable customers to use the preview feature on .NET 6 anymore. However, for features like HTTP3 (which doesn't require tooling features) this might not apply. So I guess it's case by case.
